src/boot/efi/addon.c

```diff
@@ -0,0 +1,10 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "efi.h"
+
+/* This is intended to carry data, not to be executed */
+
+EFIAPI EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *system_table);
+EFIAPI EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *system_table) {
+    return EFI_SECURITY_VIOLATION;
+}
```

Comment:
hmm, if you add this, I think it would make sense to add the `.sdmagic` section as well, so that we clearly can discern this stub from the real stub and from sd-boot.

Reply:
still, I think all our PE binaries should carry .sdmagic, so that we can discern them. It's cheap, it's free, it's simply and our tools look for it. Let's make it easy to categorize EFI binaries we create.

Reply:
added